src/spikeinterface/sortingcomponents/tests/test_benchmark_matching.py

```diff
@@ -0,0 +1,144 @@
+import pytest
+import numpy as np
+import pandas as pd
+import shutil
+import os
+from pathlib import Path
+
+import spikeinterface.core as sc
+import spikeinterface.extractors as se
+import spikeinterface.preprocessing as spre
+from spikeinterface.sortingcomponents.benchmark import benchmark_matching
+
+@pytest.fixture(scope="session")
+def benchmark_and_kwargs():
+    recording, sorting = se.toy_example(duration=1, num_channels=2, num_units=2, num_segments=1,
+                                        firing_rate=10, seed=0)
+    recording = spre.common_reference(recording, dtype='float32')
+    cwd = Path.cwd()
+    we_path = cwd / "waveforms"
+    sort_path = cwd / "sorting.npz"
+    se.NpzSortingExtractor.write_sorting(sorting, sort_path)
+    sorting = se.NpzSortingExtractor(sort_path)
+    we = sc.extract_waveforms(recording, sorting, we_path, overwrite=True)
+    templates = we.get_all_templates()
+    noise_levels = sc.get_noise_levels(recording, return_scaled=False)
+    methods_kwargs = {
+        'tridesclous': dict(waveform_extractor=we, noise_levels=noise_levels),
+        'wobble': dict(templates=templates, nbefore=we.nbefore, nafter=we.nafter,
+                       parameters={'approx_rank' : 2})
+    }
+    methods = list(methods_kwargs.keys())
+    benchmark = benchmark_matching.BenchmarkMatching(recording, sorting, we, methods, methods_kwargs)
+    yield benchmark, methods_kwargs
+
+    # Teardown
+    shutil.rmtree(we_path)
+    os.remove(sort_path)
+
+def test_run_matching_vary_parameter(benchmark_and_kwargs):
```

Comment:
I don't think this is right... The function is called `run_matching_vary_parameter`, and I want to test that one specifically.

Reply:
I see. I probably don't understand.

Using a parameterized version will also test that the function  `bmk.run_matching_vary_parameter` does not throw an assertion with varying parameter if that's what you intend. The advantage of running parameterized test is that the CI will tell you exactly which test is failing in case one does so you can focus on what type of parameter causes the error.

Running a loop inside a test, on the other hand, means that the test will fail with the first error and then it will stop. That is, the information about the other errors will not be deliver to you unless you first fix the first one.

This is what would be gained, what is not clear to me is, what would be missed?

Reply:
My bad, I misunderstood your suggestion.  I've updated it to use pytest.mark.parameterize now